pep-9999.txt

```diff
@@ -0,0 +1,200 @@
+and other key files, and by the ``site`` module to determine the location of the
+``site-package`` directories.  Currently the prefix is found -- assuming
+``PYTHONHOME`` is not set -- by first walking up the filesystem tree looking for
+a marker file (``os.py``) that signifies the presence of the standard library,
+and if none is found, falling back to the build-time prefix hard coded in the
+binary. The result of this process is the contents of sys.path - a list of
+locations that the Python import system will search for modules.
+
+This PEP proposes to add a new step in this process. If a ``__pypackages__``
+directory is found in the current working directory, then it will be included in
+``sys.path`` after the current working directory and just before the system
+site-packages. This way, if the Python executable starts in the given project
+directory, it will automatically find all the dependencies inside of
+``__pypackages__``.
+
+In case Python scripts, Python will try to find ``__pypackages__`` in the base
+directory of the script. If found (along with the current Python version
+directory inside), then it will be used, otherwise the Python will behave as it
+does currently.
+
+
+If any package management tool finds the same ``__pypackages__`` directory in
+the current working directory, it will install any packages there and also
+create it if required based on Python version.
+
+
+Projects that use a source management system can include a ``__pypackages__``
+directory (empty or with e.g. a file like ``.gitignore``). After doing a fresh
+check out the source code, ``pip`` can be used to install the required
+dependencies directly into this directory.
+
+Example
+-------
+
+The following shows an example project directory structure, and different ways
+the Python executable and any script will behave.
+
+::
+
+    foo
+        __pypackages__
+            python3.8
+                Lib
+                    bottle
+        myscript.py
+
+    /> python foo/myscript.py
+    sys.path[0] == 'foo'
+    sys.path[x] == 'foo/__pypackages__/python3.8/Lib'
+    
+
+    cd foo
+
+    foo> /usr/bin/randomproject.sh
+        python /usr/share/randomproject/main.py
+
+    foo> /usr/bin/ansible  
+        #! /usr/bin/env python3
+    foo> python /usr/bin/ansible
+
+
+    foo> python myscript.py
+
+
+    foo> python
+    sys.path[0] == '.'
+    sys.path[x] == './__pypackages__/python3.8/Lib'
+
+    foo> python -m bottle
+
+
+    
+
+We have a project directory called ``foo`` and it has a ``__pypackages__``
+inside of it. We have ``bottle`` installed in that
+``__pypackages__/python3.8/Lib``, and have ``myscript.py`` file inside of the
+project directory.
+
+For invoking a script, for example ``/usr/bin/randomproject.sh``, Python will
+try to find a ``__pypackages__`` inside of the directory that the script
+resides[1]_, ``/usr/bin``.  The same will happen in case of the last example,
```

Comment:
The script in this example seems to actually live in `/usr/share/randomproject`; it's a shell script (not a Python script) that lives in `/usr/bin/` ?